patches/server/0989-Fire-entity-death-event-for-ender-dragon.patch

```diff
@@ -0,0 +1,33 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Trevor Bedson <78997566+Prorickey@users.noreply.github.com>
+Date: Fri, 14 Jul 2023 20:47:02 -0400
+Subject: [PATCH] Fire entity death event for ender dragon
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java b/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
+index 92666c48620078623a451fbf68f673cb9f81c4b5..0dda2487dff6685560069c68e54e076ca4dca80a 100644
+--- a/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
++++ b/src/main/java/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
+@@ -651,13 +651,21 @@ public class EnderDragon extends Mob implements Enemy {
+ 
+     @Override
+     public void kill() {
++        // Paper start
++        this.silentDeath = true;
++        org.bukkit.event.entity.EntityDeathEvent deathEvent =  org.bukkit.craftbukkit.event.CraftEventFactory.callEntityDeathEvent(this, new java.util.ArrayList<>());
++        if(deathEvent.isCancelled()) {
```

Comment:
It would also possible to remove the deathEvent variable entirely, but comes down to the same thing

```suggestion
+        final var deathEvent =  org.bukkit.craftbukkit.event.CraftEventFactory.callEntityDeathEvent(this);
+        if (deathEvent.isCancelled()) {
```

Reply:
No var